src/Avalonia.Visuals/Media/Immutable/ImmutablePen.cs

```diff
@@ -44,6 +45,8 @@ public class ImmutablePen : IPen, IEquatable<IPen>
             PenLineJoin lineJoin = PenLineJoin.Miter,
             double miterLimit = 10.0)
         {
+            Debug.Assert(!(brush is IMutableBrush));
```

Comment:
I've always tried to keep `Debug.Assert` out of our codebase because it causes popups to be shown when running unit tests and causes everything to basically hang. There are [workarounds](https://stackoverflow.com/questions/2340890/nunit-conflict-with-debug-assert) but nothing I've ever found personally satisfactory... Anyone else have opinions on this?

Reply:
Actually I just checked and it looks like this might be a solved problem in modern tooling, so I'm fine with keeping the Debug.Assert in.